scripts/list-external-dependencies.js

```diff
@@ -1,5 +1,72 @@
+const { readFileSync } = require('fs')
 const { join } = require('path')
 
+const COLORS = {
+  RESET: '\x1b[0m',
+  FG_RED: '\x1b[31m',
+  FG_GREEN: '\x1b[32m',
+  FG_YELLOW: '\x1b[33m',
+  FG_BLUE: '\x1b[34m',
+}
+
+const LOGGER = {
+  danger(message) {
+    console.log(`${COLORS.FG_RED}%s${COLORS.RESET}`, message)
+  },
+  warning(message) {
+    console.log(`${COLORS.FG_YELLOW}%s${COLORS.RESET}`, message)
+  },
+  info(message) {
+    console.log(`${COLORS.FG_BLUE}%s${COLORS.RESET}`, message)
+  },
+  success(message) {
+    console.log(`${COLORS.FG_GREEN}%s${COLORS.RESET}`, message)
+  },
+}
+
+function countExternalDependencies(dependency) {
+  try {
+    const { dependencies, devDependencies } = JSON.parse(
+      readFileSync(`./node_modules/${dependency}/package.json`).toString()
+    )
+
+    return {
+      dependencies: dependencies === undefined ? 0 : new Set(Object.keys(dependencies)).size,
+      devDependencies:
+        devDependencies === undefined ? 0 : new Set(Object.keys(devDependencies)).size,
+    }
+  } catch {
+    return {
+      dependencies: 0,
+      devDependencies: 0,
```

Comment:
I'm not clear on what instances this function would throw, but is it reasonable to just default to `0` in those cases? If not, perhaps we could default to some other placeholder value (e.g. -1, NaN, .. ) and color it differently to avoid "false negatives"?